src/OmniSharp.LanguageServerProtocol/Handlers/OmniSharpDocumentHighlightHandler.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Models;
+using OmniSharp.Models.FindUsages;
+
+namespace OmniSharp.LanguageServerProtocol.Handlers
+{
+    internal sealed class OmniSharpDocumentHighlightHandler : DocumentHighlightHandlerBase
+    {
+        public static IEnumerable<IJsonRpcHandler> Enumerate(RequestHandlers handlers)
+        {
+            foreach (var (selector, handler) in handlers
+                .OfType<Mef.IRequestHandler<FindUsagesRequest, QuickFixResponse>>())
+                if (handler != null)
+                    yield return new OmniSharpDocumentHighlightHandler(handler, selector);
+        }
+
+        private readonly Mef.IRequestHandler<FindUsagesRequest, QuickFixResponse> _findUsagesHandler;
+        private readonly DocumentSelector _documentSelector;
+
+        public OmniSharpDocumentHighlightHandler(Mef.IRequestHandler<FindUsagesRequest, QuickFixResponse> findUsagesHandler, DocumentSelector documentSelector)
+        {
+            _findUsagesHandler = findUsagesHandler;
+            _documentSelector = documentSelector;
+        }
+
+        public override async Task<DocumentHighlightContainer> Handle(DocumentHighlightParams request, CancellationToken token)
+        {
+            // TODO: Utilize Roslyn ExternalAccess to take advantage of HighlightingService.
+
+            var omnisharpRequest = new FindUsagesRequest
+            {
+                FileName = Helpers.FromUri(request.TextDocument.Uri),
+                Column = Convert.ToInt32(request.Position.Character),
+                Line = Convert.ToInt32(request.Position.Line),
+                OnlyThisFile = true,
+                ExcludeDefinition = false
+            };
+
+            var omnisharpResponse = await _findUsagesHandler.Handle(omnisharpRequest);
```

Comment:
interesting - findusages returns a quick fix response that actually contains document highlights?  Some strange naming

Reply:
a lot of history behind it 😀